libs/traits/src/liquidity_pools.rs

```diff
@@ -12,34 +12,35 @@
 // GNU General Public License for more details.
 
 use frame_support::dispatch::{DispatchResult, DispatchResultWithPostInfo};
-use parity_scale_codec::Input;
+use sp_runtime::DispatchError;
 use sp_std::vec::Vec;
 
 /// An encoding & decoding trait for the purpose of meeting the
 /// LiquidityPools General Message Passing Format
-pub trait Codec: Sized {
+pub trait LPEncoding: Sized {
 	fn serialize(&self) -> Vec<u8>;
-	fn deserialize<I: Input>(input: &mut I) -> Result<Self, parity_scale_codec::Error>;
+	fn deserialize(input: &[u8]) -> Result<Self, DispatchError>;
 }
 
 #[cfg(any(test, feature = "std"))]
 pub mod test_util {
-	use parity_scale_codec::{Decode, Encode, Input, MaxEncodedLen};
+	use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 	use scale_info::TypeInfo;
 
-	use super::Codec;
+	use super::*;
 
 	#[derive(Debug, Eq, PartialEq, Clone, Encode, Decode, TypeInfo, MaxEncodedLen)]
 	pub struct Message;
-	impl Codec for Message {
+	impl LPEncoding for Message {
 		fn serialize(&self) -> Vec<u8> {
 			vec![0x42]
 		}
 
-		fn deserialize<I: Input>(input: &mut I) -> Result<Self, parity_scale_codec::Error> {
-			match input.read_byte()? {
-				0x42 => Ok(Self),
-				_ => Err("unsupported message".into()),
+		fn deserialize(input: &[u8]) -> Result<Self, DispatchError> {
+			match input.first() {
+				Some(0x42) => Ok(Self),
```

Comment:
Nit - can we also have a const for this value and use that in the tests?

Reply:
In which places in the test are you referring?

The only place where we use that value (but as a string) is here: https://github.com/centrifuge/centrifuge-chain/pull/1891/files#diff-0f02c17540413a2c2eb2d6554089fac0df041bbc8f248e5a3945b600761f0fe4R409 (which is true that with some work we can tied both)

In the rest of the tests what we do is check the serialization and deserialization matches.

Reply:
Actually, the constant is not needed: [4cc4899](https://github.com/centrifuge/centrifuge-chain/pull/1889/commits/4cc4899bb265f3ff6be747a044700e4ebbd71de8)

Reply:
Saw [this](https://github.com/centrifuge/centrifuge-chain/blob/exp/lp-message-bincode/pallets/liquidity-pools-gateway/routers/src/tests.rs#L347) and thought that a different value here might affect the test?